### PR TITLE
Cypress/E2E: Fix public links and channel mode

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Fix public links and channel mode

#### Ticket Link
none, failed on daily run

#### Screenshots
Screen Shot 2021-11-24 at 6 26 08 PM
Screen Shot 2021-11-24 at 6 27 22 PM
Screen Shot 2021-11-24 at 6 50 32 PM

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: e2e/cypress/utils/constants.js [hunk 1]
```diff
@@ -13,6 +13,11 @@ export const CLOUD = 'Cloud';
 export const E20 = 'E20';
 export const TEAM = 'Team';
 
+export const FixedPublicLinks = {
```
Comment (metanerd) [inline e2e/cypress/utils/constants.js:16]: :thinking: What does "fixed" mean in this context? Static or it is a temporary fix? Anyway, this is non blocking. I think I just don't understand.

**Review by metanerd — APPROVED** [review]
Thank you for taking care @saturninoabril !

File: e2e/cypress/integration/enterprise/ldap_group/channel_modes_spec.js [hunk 2]
```diff
@@ -73,10 +73,16 @@ describe('LDAP Group Sync - Test channel public/private toggle', () => {
...
             cy.visit(`/admin_console/user_management/channels/${id}`);
             cy.get('#channel_profile').contains('Town Square');
-            cy.get('#channel_manage .group-teams-and-channels--body').find('button').eq(0).should('have.class', 'false');
-            cy.get('#channel_manage .group-teams-and-channels--body').find('button').eq(1).contains('Public');
-            cy.get('#channel_manage .group-teams-and-channels--body').find('button').eq(0).should('have.class', 'disabled');
-            cy.get('#channel_manage .group-teams-and-channels--body').find('button').eq(1).should('have.class', 'disabled');
+            cy.get('#channel_manage').scrollIntoView().should('be.visible').within(() => {
+                cy.get('.line-switch').first().within(() => {
+                    cy.findByText('Sync Group Members').should('be.visible');
+                    cy.get('#syncGroupSwitch').should('be.disabled');
```
Comment (Phrynobatrachus) [inline e2e/cypress/integration/enterprise/ldap_group/channel_modes_spec.js:79]: not sure if I'm missing something but this `get` is failing for me, and I don't see a normal ID with that name in the DOM. there are however a couple `data-testid`s there.

I had to change this to `cy.findByTestId('syncGroupSwitch-button')` to pass

File: e2e/cypress/integration/enterprise/ldap_group/channel_modes_spec.js [hunk 3]
```diff
@@ -73,10 +73,16 @@ describe('LDAP Group Sync - Test channel public/private toggle', () => {
...
+                });
+                cy.get('.line-switch').last().within(() => {
+                    cy.findByText('Public channel or private channel').should('be.visible');
+                    cy.get('#allow-all-toggle').should('be.disabled');
```
Comment (Phrynobatrachus) [inline e2e/cypress/integration/enterprise/ldap_group/channel_modes_spec.js:83]: similarly here, had to change to `cy.findByTestId('allow-all-toggle-button')` to get this assertion to work

**Review by Phrynobatrachus — CHANGES_REQUESTED** [review]

File: e2e/cypress/utils/constants.js (hunk 1 above)
Comment (saturninoabril) [reply e2e/cypress/utils/constants.js:16]: It's hard to name things :) but in this context, the public links are hard-coded in our codebase. Other instances of terms of service and privacy policy links in the app are configurable via server config but not in "Edition and License" at system console page and some section of the app. Hence, I prefix it with "Fixed" but other name would work too. 0/5 on naming and very much open to more appropriate suggestion.

File: e2e/cypress/integration/enterprise/ldap_group/channel_modes_spec.js (hunk 2 above)
Comment (saturninoabril) [reply e2e/cypress/integration/enterprise/ldap_group/channel_modes_spec.js]: Thanks Cass! Not sure why but it works for me. Anyway, I changed per your suggestions and it passed on me as well. Please take a look again.

File: e2e/cypress/integration/enterprise/ldap_group/channel_modes_spec.js (hunk 3 above)
Comment (saturninoabril) [reply e2e/cypress/integration/enterprise/ldap_group/channel_modes_spec.js]: Same here as commented above.

**Review by Phrynobatrachus — APPROVED** [review]

**Pull request merged**